reward/stakingInfo.go

```diff
@@ -0,0 +1,152 @@
+	CouncilRewardAddrs  []common.Address // Address of Council account which will get block reward
+	KIRAddr             common.Address   // Address of KIR contract
+	PoCAddr             common.Address   // Address of PoC contract
+
+	UseGini bool
+	Gini    float64 // gini coefficient
+
+	// Derived from CouncilStakingAddrs
+	CouncilStakingAmounts []uint64 // Staking amounts of Council
+}
+
+func newEmptyStakingInfo(blockNum uint64) (*StakingInfo, error) {
+	stakingInfo := &StakingInfo{
+		BlockNum:              blockNum,
+		CouncilNodeAddrs:      make([]common.Address, 0, 0),
+		CouncilStakingAddrs:   make([]common.Address, 0, 0),
+		CouncilRewardAddrs:    make([]common.Address, 0, 0),
+		KIRAddr:               common.Address{},
+		PoCAddr:               common.Address{},
+		CouncilStakingAmounts: make([]uint64, 0, 0),
+		Gini:                  DefaultGiniCoefficient,
+		UseGini:               false,
+	}
+	return stakingInfo, nil
+}
+
+func newStakingInfo(bc *blockchain.BlockChain, helper governanceHelper, blockNum uint64, nodeIds []common.Address, stakingAddrs []common.Address, rewardAddrs []common.Address, KIRAddr common.Address, PoCAddr common.Address) (*StakingInfo, error) {
+	intervalBlock := bc.GetBlockByNumber(blockNum)
+	statedb, err := bc.StateAt(intervalBlock.Root())
+	if err != nil {
+		logger.Trace("Failed to make a state for interval block", "interval blockNum", blockNum, "err", err)
+		return nil, err
+	}
+
+	// Get balance of stakingAddrs
+	stakingAmounts := make([]uint64, len(stakingAddrs))
+	for i, stakingAddr := range stakingAddrs {
+		tempStakingAmount := big.NewInt(0).Div(statedb.GetBalance(stakingAddr), big.NewInt(0).SetUint64(params.KLAY))
+		if tempStakingAmount.Cmp(maxStakingLimitBigInt) > 0 {
+			tempStakingAmount.SetUint64(maxStakingLimit)
+		}
+		stakingAmounts[i] = tempStakingAmount.Uint64()
+	}
+
+	var useGini bool
+	if res, err := helper.GetItemAtNumberByIntKey(blockNum, params.UseGiniCoeff); err != nil {
+		logger.Trace("Failed to get useGiniCoeff from governance", "blockNum", blockNum, "err", err)
+		return nil, err
+	} else {
+		useGini = res.(bool)
+	}
+	gini := DefaultGiniCoefficient
+
+	stakingInfo := &StakingInfo{
+		BlockNum:              blockNum,
+		CouncilNodeAddrs:      nodeIds,
+		CouncilStakingAddrs:   stakingAddrs,
+		CouncilRewardAddrs:    rewardAddrs,
+		KIRAddr:               KIRAddr,
+		PoCAddr:               PoCAddr,
+		CouncilStakingAmounts: stakingAmounts,
+		Gini:                  gini,
+		UseGini:               useGini,
+	}
+	return stakingInfo, nil
+}
+
+func (s *StakingInfo) GetIndexByNodeId(nodeId common.Address) int {
+	for i, addr := range s.CouncilNodeAddrs {
+		if addr == nodeId {
+			return i
+		}
+	}
+	return AddrNotFoundInCouncilNodes
+}
+
+func (s *StakingInfo) GetStakingAmountByNodeId(nodeId common.Address) uint64 {
+	i := s.GetIndexByNodeId(nodeId)
+	if i != AddrNotFoundInCouncilNodes {
+		return s.CouncilStakingAmounts[i]
+	}
+	return 0
+}
+
+type uint64Slice []uint64
```

Comment:
I think that `[]uint64` is better. Do you have any reason to redefine this type?